src/pegen/parser.py

```diff
@@ -111,7 +111,11 @@ def memoize_left_rec_wrapper(self: P) -> Optional[T]:
 
             while True:
                 self._reset(mark)
-                result = method(self)
+                self.in_recursive_rule += 1
+                try:
+                    result = method(self)
+                finally:
+                    self.in_recursive_rule -= 1
```

Comment:
This would look a bit better as a context manager, but I think this is fine in any case

Reply:
If you want, create another PR for transforming this into a context manager, but is ok if you don't want to :)

Reply:
Since this shows up in a single place and is not meant to be used elsewhere I think the try finally is fine, otherwise I would agree.